sdk/search/Azure.Search.Documents/src/Indexes/Models/SearchableField.cs

```diff
@@ -51,7 +54,19 @@ public SearchableField(string name, bool collection = false) : base(name, collec
         /// Assigning a synonym map to a field ensures that query terms targeting that field are expanded at query-time using the rules in the synonym map.
         /// This attribute can be changed on existing fields.
         /// </remarks>
-        public IList<string> SynonymMapNames { get; } = new List<string>();
+        public IList<string> SynonymMapNames
+        {
+            get => _synonymMapNames;
+            internal set
+            {
+                _synonymMapNames.Clear();
```

Comment:
I wasn't really sure why we're making this change.  Is it needed for FieldBuilder?

Reply:
Needed, no. Convenient while consolidated the logic into the target component, yes. I'd have to put the same logic (well, sans `Clear`) elsewhere, and figured it works better here.